googlemock/include/gmock/gmock-spec-builders.h

```diff
@@ -1790,10 +1791,74 @@ void ReportUninterestingCall(CallReaction reaction, const std::string& msg);
 
 }  // namespace internal
 
-// A MockFunction<F> class has one mock method whose type is F.  It is
-// useful when you just want your test code to emit some messages and
-// have Google Mock verify the right messages are sent (and perhaps at
-// the right times).  For example, if you are exercising code:
+// The SignatureOf<F> struct is a meta-function returning function
+// signature corresponding to the provided F argument. It makes use of
+// MockFunction easier by allowing it to accept more F arguments than
+// just function signatures.  Specializations provided here cover only
+// a signature type itself and std::function. If a user wishes to use
+// SignatureOf with other types (like for example boost::function)
+// a corresponding specialization must be provided.
+template <typename F>
+struct SignatureOf;
+
+template <typename R, typename... Args>
+struct SignatureOf<R(Args...)> {
+  using type = R(Args...);
+};
+
+template <typename R, typename... Args>
+struct SignatureOf<std::function<R(Args...)>> {
+  using type = R(Args...);
+};
+
+template <typename F>
+using SignatureOfT = typename SignatureOf<F>::type;
+
+namespace internal {
+
+template <typename F>
+class MockFunction;
+
+template <typename R, typename... Args>
+class MockFunction<R(Args...)> {
+ public:
+  MockFunction() = default;
+  MockFunction(const MockFunction&) = delete;
+  MockFunction& operator=(const MockFunction&) = delete;
+
+  std::function<R(Args...)> AsStdFunction() {
+    return [this](Args... args) -> R {
+      return this->Call(std::forward<Args>(args)...);
```

Comment:
This and other std::forwards won't forward anything. `args` can never be variadic universal references here as they are not function template parameters. Imo, std::move works better here (as it will try to move, unless not possible (l-refs and nonmovable types)).

Reply:
Firstly, note that I haven't written it so. I only moved it a bit. It was so before.

Secondly, I think you are not right about it anyway. This was originally `std::move` back in time when the class was still located in `gmock-generated-function-mockers.h` but was corrected to `std::forward` by 7796273 related to #1867.

Reply:
Sorry then, thanks for a good reference to another PR.